### PR TITLE
Setup E2E test runner for react native sample applications

### PR DESCRIPTION
**Pull request by oleggnidets-okta**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+[![CI Status](https://travis-ci.com/okta/samples-js-react-native.svg?branch=master)](https://travis-ci.com/github/okta/samples-js-react-native)
```
Comment (shuowu) [inline README.md:1]: 👍

File: custom-sign-in/.eslintrc.js [hunk 2]
```diff
@@ -1,4 +1,86 @@
 module.exports = {
-  root: true,
-  extends: '@react-native-community',
+  ignorePatterns: [
```
Comment (shuowu) [inline custom-sign-in/.eslintrc.js:2]: Let's follow the pattern to separate eslint files into different folders, like one for root, one in src, one in test. 
Here is how we added it in [okta-react-repo](https://github.com/okta/okta-react/tree/master/test/jest).

File: custom-sign-in/App.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * Copyright (c) 2019-Present, Okta, Inc. and/or its affiliates. All rights reserved.
```
Comment (shuowu) [inline custom-sign-in/App.js:2]: check `yarn banners` script, the banners can be updated automatically.

File: custom-sign-in/app/components/Error.js [hunk 4]
```diff
@@ -1,12 +1,24 @@
...
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
 import React from 'react';
 import { StyleSheet, Text } from 'react-native';
 
 const Error = ({ error }) => {
-  if (!error) {
+  if (!error || error == '') {
```
Comment (shuowu) [inline custom-sign-in/app/components/Error.js:17]: In js, `!error` means error may equal to `0`, `empty string`, `null`, `undefined`, very dynamic, hah.

File: custom-sign-in/samples.config.js [hunk 5]
```diff
@@ -10,13 +10,15 @@
...
 
+import { CLIENT_ID, REDIRECT_URI, LOGOUT_REDIRECT_URI, ISSUER } from '@env';
+
 export default {
   oidc: {
-    clientId: '{clientId}',
-    redirectUri: '{customUriScheme}:/loginCallback',
-    endSessionRedirectUri: '{customUriScheme}:/logoutCallback',
-    discoveryUri: 'https://{yourOktaDomain}/oauth2/default',
+    clientId: CLIENT_ID,
```
Comment (shuowu) [inline custom-sign-in/samples.config.js:17]: What's the behavior before? Devs need to change the config manually? 
How about also add comments to explain what each config should look like?

**Review by shuowu — COMMENTED** [review]
Thanks for the PR! JS parts generally LGTM.

File: browser-sign-in/ios/browserSignIn.xcodeproj/xcshareddata/xcschemes/browserSignIn.xcscheme [hunk 6]
```diff
@@ -51,22 +51,21 @@
...
-      </Testables>
+      shouldUseLaunchSchemeArgsEnv = "YES"
+      language = "en"
+      region = "US">
+      <PreActions>
+         <ExecutionAction
+            ActionType = "Xcode.IDEStandardExecutionActionsCore.ExecutionActionType.ShellScriptAction">
+            <ActionContent
+               title = "Run Script"
+               scriptText = "defaults write com.apple.iphonesimulator ConnectHardwareKeyboard 0&#10;">
```
Comment (mikenachbaur-okta) [inline browser-sign-in/ios/browserSignIn.xcodeproj/xcshareddata/xcschemes/browserSignIn.xcscheme:65]: 👍

File: browser-sign-in/ios/browserSignInUITests/BrowserSignInUITests.swift [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+final class BrowserSignInUITests: XCTestCase {
+  
+  private var username = ProcessInfo.processInfo.environment["USERNAME"]!
+  private var password = ProcessInfo.processInfo.environment["PASSWORD"]!
+  
+  private var app: XCUIApplication!
+  private let springboard = XCUIApplication(bundleIdentifier: "com.apple.springboard")
+  
+  private var loginButton: XCUIElement {
```
Comment (mikenachbaur-okta) [inline browser-sign-in/ios/browserSignInUITests/BrowserSignInUITests.swift:23]: nice!

**Review by mikenachbaur-okta — APPROVED** [review]
The Swift sections LGTM, I like that the UI tests are integrated into Travis, and are passing.

File: custom-sign-in/App.js (hunk 3 above)
Comment (oleggnidets-okta) [reply custom-sign-in/App.js:2]: Screenshot 2021-03-23 at 16 17 27

File: custom-sign-in/app/components/Error.js (hunk 4 above)
Comment (oleggnidets-okta) [reply custom-sign-in/app/components/Error.js:17]: Fixed :) 
It reminded me a JS "tutorial" 😄 
https://www.youtube.com/watch?v=et8xNAc2ic8

File: browser-sign-in/ios/browserSignIn.xcodeproj/xcshareddata/xcschemes/browserSignIn.xcscheme (hunk 6 above)
Comment (oleggnidets-okta) [reply browser-sign-in/ios/browserSignIn.xcodeproj/xcshareddata/xcschemes/browserSignIn.xcscheme:65]: Interesting fact: `typeText` works from time to time when Simulator has enabled option `Connected Hardware Keyboard` ¯\_(ツ)_/¯

File: browser-sign-in/ios/browserSignInUITests/BrowserSignInUITests.swift (hunk 7 above)
Comment (oleggnidets-okta) [reply browser-sign-in/ios/browserSignInUITests/BrowserSignInUITests.swift:23]: Thanks!

File: custom-sign-in/App.js (hunk 3 above)
Comment (shuowu) [reply custom-sign-in/App.js:2]: looks like `globby` is the root level deps of the repo. In the root folder try `rm -rf node_modules` -> `npm install` -> `npm run banners`. It works for me locally on master.

**Review by shuowu — APPROVED** [review]

File: custom-sign-in/samples.config.js (hunk 5 above)
Comment (oleggnidets-okta) [reply custom-sign-in/samples.config.js]: Added

**Pull request closed without merging**